### PR TITLE
Feat new functionalities

### PR DESCRIPTION
**Pull request by lohanidamodar**
Added functionalities and tests.

### CODE REVIEW

**Comment by eldadfux** [comment]
@lohanidamodar please add a few tests that uses both resize and the new options, just to make sure behavior is as expected.

**Comment by lohanidamodar** [comment]
@eldadfux will do.

**Comment by eldadfux** [comment]
@lohanidamodar while we wait for you to discuss the border problems with @christyjacob4, please add more examples to the readme showing some of the new methods in action.

**Comment by lohanidamodar** [comment]
@eldadfux Have updated the Readme, as well as fixed the issue, however, still need to make sure that assertions are enough so @christyjacob4 and @TorstenDittmann could also provide feedback on regarding my assertions.

File: tests/Image/ImageTest.php [hunk 1]
```diff
@@ -167,4 +167,122 @@ public function testCrop100x100GIF()
...
+        $this->assertEquals(\is_readable($target), true);
+        $this->assertEquals(\mime_content_type($target), \mime_content_type($original));
+        $this->assertNotEmpty(\md5(\file_get_contents($target)));
+
+        $image = new \Imagick($target);
+        $this->assertEquals('JPEG', $image->getImageFormat());
+        \unlink($target);
+    }
+
+    public function testRotate45()
```
Comment (TorstenDittmann) [inline tests/Image/ImageTest.php:190]: For this you could take a non square image, rotate it and check the resolution afterwerds 👍🏻

File: tests/Image/ImageTest.php [hunk 2]
```diff
@@ -167,4 +167,122 @@ public function testCrop100x100GIF()
...
+        $this->assertEquals(\is_readable($target), true);
+        $this->assertEquals(\mime_content_type($target), \mime_content_type($original));
+        $this->assertNotEmpty(\md5(\file_get_contents($target)));
+
+        $image = new \Imagick($target);
+        $this->assertEquals('PNG', $image->getImageFormat());
+        \unlink($target);
+    }
+
+    public function testCrop100Op05()
```
Comment (TorstenDittmann) [inline tests/Image/ImageTest.php:247]: I would test for the image resolution here 🙂

**Review by TorstenDittmann — CHANGES_REQUESTED** [review]
Added an idea for the tests 👍🏻

**Comment by lohanidamodar** [comment]
Thank you @TorstenDittmann I will try

**Review by eldadfux — APPROVED** [review]

**Review by TorstenDittmann — APPROVED** [review]

**Pull request merged**